src/site/_transforms/width-height-images.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview An eleventy transform that makes sure all local images have
+ * width and height html attributes set to avoid layout shift.
+ */
+
+const cheerio = require('cheerio');
+const sizeOf = require('image-size');
+const path = require('path');
+const fs = require('fs');
+
+module.exports = (content, outputPath) => {
+  if (!outputPath || !outputPath.endsWith('.html')) {
+    return content;
+  }
+
+  const $ = cheerio.load(content);
+  $('img').each((_, elem) => {
+    const $elem = $(elem);
+    const originalSrc = $elem.attr('src').trim();
+    const isLocal = !RegExp('^(https?://|/)').test(originalSrc);
+    if (!originalSrc || !isLocal) {
+      return;
+    }
```

Comment:
There's a transform that runs before this one called `responsive-images` which sets every image to come from our image CDN. As a result, this section of code causes the transform to opt-out of measuring most images.

![image](https://user-images.githubusercontent.com/1066253/102152878-d18a9180-3e2a-11eb-9164-95b6b2010b58.png)

I think you'd want to run your transform either before responsive-images, or perhaps as part of responsive-images (it could be a module that exports a function which expects the cheerio object). That way we wouldn't need to parse each page twice with cheerio.

Reply:
My latest patch hopefully addresses your feedback by merging this transform into responsive-images.